src/test/java/uk/gov/hmcts/reform/ccd/document/am/feign/CaseDocumentClientApiTest.java

```diff
@@ -0,0 +1,312 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@SpringBootTest(classes = {CaseDocumentClientApi.class})
+@PropertySource(value = "classpath:application.yml")
+@EnableAutoConfiguration
+@AutoConfigureWireMock(port = 5050)
+class CaseDocumentClientApiTest {
+
+    private static final UUID DOCUMENT_ID = UUID.randomUUID();
+
+    private static final String URL = "/cases/documents";
+
+    private static final String CLASSIFICATION = "classification";
+    private static final String CASE_TYPE_ID = "CaseTypeID";
+
+    private static final String SERVICE_AUTHORISATION_KEY = "ServiceAuthorization";
+    private static final String BEARER = "Bearer";
+    private static final String TOKEN = "user1";
+
+    private static final boolean PERMANENT = false;
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Autowired
+    private CaseDocumentClientApi caseDocumentClientApi;
+
+    @BeforeAll
+    static void beforeAll(){
+    }
+
+    @BeforeEach
+    void setUp() {
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+
+    }
+
+    @AfterAll
+    static void tearDown() {
+    }
+
+    @Test
+    void uploadSingleDocumentTest() throws IOException {
+        List<MultipartFile> files = new ArrayList<>();
+        files.add(getMultiPartFile("testFile1"));
+
+        ResponseEntity response = new ResponseEntity(HttpStatus.OK);
+
+        stubForUpload(response);
+
+        ResponseEntity finalResponse = caseDocumentClientApi.uploadDocuments(
+            files,
+            CLASSIFICATION,
+            CASE_TYPE_ID,
+            BEARER + TOKEN
+        );
+
+        assertEquals(HttpStatus.OK, finalResponse.getStatusCode());
+        assertEquals(1, files.size());
+    }
+
+    @Test
+    void uploadMultipleDocumentTest() throws IOException {
+        List<MultipartFile> files = new ArrayList<>();
+        files.add(getMultiPartFile("testFile1"));
+        files.add(getMultiPartFile("testFile2"));
+        files.add(getMultiPartFile("testFile3"));
+
+        ResponseEntity response = new ResponseEntity(HttpStatus.OK);
+
+        stubForUpload(response);
```

Comment:
ideally pass all request params to stubForUpload method and match all of them correctly. If your response is just plain http OK, then no need to pass here. Inside stubForUpload you can do just `    .willReturn(aResponse()
                                    .withStatus(HttpStatus.OK.value())`